### PR TITLE
Refactor zoom link sync further.

### PR DESCRIPTION
**Pull request by peterdeweese**
# Description

This includes many small refactors and resolves most rubocop offenses that require `rubocop -A` or manual work for zoom link sync. 10 offenses are left but more importantly, our rubocop.yml file is improved.

# Motivation and Context
To learn about the sync process for systems continuity.
To improve linting results and consistency to make the code more consistent and understandable.
To improve our ability to maintain and add features in this code.
To discuss code features with the team and to pair.

Task: 
https://app.asana.com/0/1201131148207877/1202848766806841/f

### CODE REVIEW

File: app/models/participant_sync_info.rb [hunk 1]
```diff
@@ -23,17 +23,17 @@ def self.primary_key
   belongs_to :user, primary_key: :salesforce_contact_id, foreign_key: :contact_id
 
   belongs_to :accelerator_course,
-             primary_key: :canvas_course_id, foreign_key: :canvas_accelerator_course_id, class_name: Course.name
+             primary_key: :canvas_course_id, foreign_key: :canvas_accelerator_course_id, class_name: 'Course'
```
Comment (peterdeweese) [inline app/models/participant_sync_info.rb:26]: I learned the hard way once that using external class info inside classes can occasionally cause problems during class load. Rubocop has a handy warning for this.

File: app/models/participant_sync_info.rb [hunk 2]
```diff
@@ -107,7 +107,7 @@ def self.included(base)
             .where.not('participant__c.status__c': nil) # ignore folks where we cleared their Status
         }
       end
-    end # self.included
+    end
```
Comment (peterdeweese) [inline app/models/participant_sync_info.rb:110]: Rubocop has a warning for any comments on the same line that is resolved with a `-A` run. I didn't notice these were removed until looking at the diff here in GitHub. I always use folding, vertical lines from the IDE, or the indicator when you put the cursor here to see which do or other beginning token matches that is in Atom or VSC. Not sure if others depend on these.

File: app/models/participant_sync_info.rb [hunk 3]
```diff
@@ -116,7 +116,7 @@ def self.included(base)
     # ]
     # Access this blob using the 'ta_caseload_enrollments' column like
     # HerokuConnect::Participant.live_sf_data.find_by(some_condition).ta_caseload_enrollments
-    TA_CASELOAD_ENROLLMENTS_SQL_SUBQUERY = <<~SQL
+    TA_CASELOAD_ENROLLMENTS_SQL_SUBQUERY = <<~SQL.squish
```
Comment (peterdeweese) [inline app/models/participant_sync_info.rb:119]: I think this squish removes the extra whitespace we use to format the file. Again a `-A` improvement.

File: app/models/participant_sync_info.rb [hunk 4]
```diff
@@ -170,43 +170,43 @@ def self.diffs_for_program(program)
     # Create the diffs. Note that last_sync_info can be nil
     new_sync_infos.map do |new_sync_info|
       last_sync_info = last_sync_infos[new_sync_info.sfid]
-      pdiff = ParticipantSyncInfo::Diff.new(last_sync_info, new_sync_info)
+      ParticipantSyncInfo::Diff.new(last_sync_info, new_sync_info)
     end
   end
 
-  def is_enrolled?
+  def enrolled?
```
Comment (peterdeweese) [inline app/models/participant_sync_info.rb:177]: A prefix like `is` or `has` for boolean queries is conventional in some languages like Java, Swift, and Objective-C, but not in Ruby.

File: app/models/participant_sync_info.rb [hunk 5]
```diff
@@ -170,43 +170,43 @@ def self.diffs_for_program(program)
...
+  def dropped?
     status == HerokuConnect::Participant::Status::DROPPED
   end
 
-  def is_withdrew?
+  def withdrew?
     status == HerokuConnect::Participant::Status::WITHDREW
   end
 
   def left_course?
```
Comment (peterdeweese) [inline app/models/participant_sync_info.rb:191]: It's worth noting that we weren't consistent with `is` either.

File: app/services/sync_zoom_links_for_participant.rb [hunk 6]
```diff
@@ -111,8 +111,10 @@ def register_participant(new_zoom_info)
     Honeycomb.start_span(name: 'sync_zoom_links_for_participant.register_participant') do
       Honeycomb.add_field('zoom_link_info.meeting_id', new_zoom_info.meeting_id)
       @participant_sync_info.add_to_honeycomb_span
-      Rails.logger.debug("  - Generating new #{new_zoom_info.salesforce_meeting_id_attribute} for " +
-                         "'#{new_zoom_info.first_name} #{new_zoom_info.last_name}'")
+      Rails.logger.debug do
```
Comment (peterdeweese) [inline app/services/sync_zoom_links_for_participant.rb:114]: String interpolation is preferred over concatenation. Among other things, it is more efficient.

File: docker-compose.yml [hunk 7]
```diff
@@ -80,7 +80,7 @@ services:
 
   platform_chrome:
     # For M1 processors: image: seleniarm/standalone-chromium
-    image: seleniarm/standalone-chromium
+    image: selenium/standalone-chrome
```
Comment (peterdeweese) [inline docker-compose.yml:83]: I accidentally checked in `seleniarm` in the prior PR. This corrects it.

File: Gemfile.lock [hunk 8]
```diff
@@ -551,7 +553,6 @@ GEM
 
 PLATFORMS
   aarch64-linux
-  arm64-darwin-22
```
Comment (peterdeweese) [inline Gemfile.lock:554]: I accidentally checked in these lock changes with my last PR. This reverts them, but I'm probably playing a tug-of-war that doesn't matter as long as we leave aarch64-linux in place.

File: spec/services/sync_zoom_links_for_participant_spec.rb [hunk 9]
```diff
@@ -288,49 +290,49 @@
             run_sync
           end
         end
-      end # END 'Enrolled Participant' examples
+      end
 
-      context 'Fellow' do
+      context 'when Fellow' do
```
Comment (peterdeweese) [inline spec/services/sync_zoom_links_for_participant_spec.rb:295]: The concept of a concept is for things that can be described with a prefix like "with", "when", etc.

File: app/services/sync_zoom_links_for_participant.rb [hunk 10]
```diff
@@ -225,27 +227,25 @@ def create_or_update_link_infos(registrant1, registrant2)
...
+				error: #{e.message}. Deleting ZoomLinkInfos with registrant_ids: #{registrant_ids}"
+      )
+      existing_zoom_link_infos.destroy_all
+      # Adds in upsert data for the second participant when there are duplicates because
+      # it deletes ZoomLinkInfo already in the table for the duplicate registrant_id
+      # so that it can add in the data here. Sending the alert above to make it known
+      # there might be a duplicate participant.
+      ZoomLinkInfo.upsert_all(
+        upsert_data, unique_by: %i[salesforce_participant_id salesforce_meeting_id_attribute]
+      )
```
Comment (tomlovett) [inline app/services/sync_zoom_links_for_participant.rb:248]: I love this kind of refactoring. Great work!

File: spec/services/sync_zoom_links_for_participant_spec.rb (hunk 9 above)
Comment (tomlovett) [reply spec/services/sync_zoom_links_for_participant_spec.rb:295]: These changes are really good improvements

**Review by tomlovett — APPROVED** [review]
Great work! Lots of valuable improvements!

**Pull request merged**